### ISSUE TITLE
Inspect: Async_Options::add_to_plugin_page() calls add_action() with a null hook on admin-ajax (PHP 8.5 deprecation flood)

### ISSUE THREAD

**Issue by HFigarella** [Automattic/jetpack#52424]
### Impacted plugin

Inspect

### Quick start

Jetpack Inspect 2.0.0-alpha (trunk, build 35163429039), PHP 8.5 (also reproduces on 8.4 without the message), WordPress 7.2 nightly.

### What I expected

`Async_Options::setup()` on `admin_init` should only hook the options script when the plugin page hook exists.

### What happened

On every `admin-ajax.php` request (Action Scheduler runner, Wordfence scan, any plugin's ajax) PHP 8.5 logs:

```
PHP Deprecated: Using null as an array offset is deprecated, use an empty string instead in wp-includes/plugin.php on line 125
```

Stack:

```
wp-includes/plugin.php:125 add_filter()
wp-includes/plugin.php:451 add_action()
jetpack-inspect/packages/Async_Option/Async_Options.php:55 add_to_plugin_page()
jetpack-inspect/packages/Async_Option/Async_Options.php:76 setup()
jetpack-inspect/options.php:40 {closure} (admin_init)
```

`get_plugin_page_hook( 'jetpack-inspect', 'admin' )` returns `null` in the admin-ajax context (no menu is registered there), and `add_to_plugin_page()` passes that straight to `add_action()`, so core does `$wp_filter[ null ]`. On three sites this produced ~2,900 deprecation entries in 12 hours.

### Steps to reproduce

1. PHP 8.5, Jetpack Inspect active, `WP_DEBUG_LOG` on.
2. Load `wp-admin/admin-ajax.php?action=heartbeat` (or let any cron/ajax run).
3. See the deprecation above.

### Suggested fix

```php
public function add_to_plugin_page( $plugin_page, $parent_page ) {
    $plugin_page_hook = get_plugin_page_hook( $plugin_page, $parent_page );
    if ( ! $plugin_page_hook ) {
        return;
    }
    add_action( $plugin_page_hook, array( $this, '_print_options_script_tag' ) );
}
```

The same `Async_Option` package is vendored in other Jetpack plugins (Boost), so the fix belongs in the package.

### Site owner impact

Log noise only, but at every ajax request; PHP 9 will turn it into an error.
